bpf/lib/strict.h

```diff
@@ -0,0 +1,60 @@
+/* SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause) */
+/* Copyright Authors of Cilium */
+
+#ifndef __STRICT_H_
+#define __STRICT_H_
+
+#include <bpf/ctx/ctx.h>
+#include <bpf/api.h>
+#include <linux/tcp.h>
+#include <linux/ip.h>
+#include <linux/ipv6.h>
+
+#include "common.h"
+#include "dbg.h"
+#include "overloadable.h"
+
+/* strict_allow checks whether the packet is allowed to pass through the strict mode. */
+static __always_inline bool
+strict_allow(struct __ctx_buff *ctx)
+{
+	void *data, *data_end;
+	__u16 proto = 0;
+	bool __maybe_unused in_strict_cidr = false;
+#ifdef ENABLE_IPV4
+	struct iphdr *ip4;
+#endif
```

Comment:
please arrange the variable definitions into reverse xmas tree (longest to shortest)
and leave a new line between them and the code